modules/uid2IdSystem.md

```diff
@@ -1,264 +1,84 @@
-## UID2 User ID Submodule
+# UID2 User ID Submodule
 
-The UID2 module handles storing, providing, and optionally refreshing tokens. While initial tokens traditionally required server-side generation, the introduction of the *Client-Side Token Generation (CSTG)* mode offers publishers the flexibility to generate UID2 tokens directly from the module, eliminating this need. Publishers can choose to operate the module in one of three distinct modes: *Client Refresh* mode, *Server Only* mode and *Client-Side Token Generation* mode.
+## Overview
 
-*Server Only* mode was originally referred to as *legacy mode*, but it is a popular mode for new integrations where publishers prefer to handle token refresh server-side.
-
-*Client-Side Token Generation* mode is included in UID2 module by default. However, it's important to note that this mode is created and made available recently. For publishers who do not intend to use it, you have the option to instruct the build to exclude the code related to this feature:
-
-```
-    $ gulp build --modules=uid2IdSystem --disable UID2_CSTG
-```
-If you do plan to use Client-Side Token Generation (CSTG) mode, please consult the UID2 Team first as they will provide required configuration values for you to use (see the Client-Side Token Generation (CSTG) mode section below for details)
+The UID2 module handles storing, providing, and optionally generating and refreshing tokens. For more details, see the [UID2 Integration Overview for Prebid.js](https://unifiedid.com/docs/guides/integration-prebid).
```

Comment:
Providing and generating are the same surely? Not sure why this part was changed. And surely storing is a moot point if generating doesn't happen, so generating can't be optional and storing not? Not totally sure that this sentence should say though.

Second sentence, just say: For details, see [publication name and link].

Reply:
I checked with Lionell and he said "[Providing] refers to providing the token to anything that requests it - user ID module, auctions, possibly other modules (like Secure Signals)", so it's distinct from generating a token.

Generating is optional because generating can be done outside of the module, e.g. server-side, then passed to the module, where it's stored in the browser and provided to other modules / auctions.

I think we want to list the four functions - storing, providing, generating, and refreshing - but I don't think it's necessary to separate the functions into optional and non-optional. I don't have a good suggestions at the moment.

Reply:
Hi @mcollins-ttd understood. Here are a couple of suggestions from me. "Providing" is unclear to me but is probably clear to the target audience. If you want to stick with what you had, no problem at all. Suggestions below and my preference is #2 but totally up to you.
Original:
The UID2 module handles storing, providing, and optionally generating and refreshing tokens.
Suggestions:
1) The UID2 module handles storing the UID2 token and providing the token to the bid stream. It can also handle generating and refreshing tokens.
2) The UID2 module can handle all aspects of managing the UID2 token: generating, storing, refreshing as needed, and sending to the bid stream.
3) The UID2 module manages the UID2 token, including storing and sending to the bid stream, as well as optionally managing token generate and token refresh.

Reply:
Replace with a list of features per discussion offline